python-avd/pyavd/_cv/client/studio_topology.py

```diff
@@ -0,0 +1,359 @@
+            }
+            request.values.append(
+                InputsConfig(
+                    key=InputsKey(
+                        studio_id=TOPOLOGY_STUDIO_ID,
+                        workspace_id=workspace_id,
+                        path=RepeatedString(values=["devices", str(device_index)]),
+                    ),
+                    inputs=json.dumps(device_entry),
+                ),
+            )
+
+        client = InputsConfigServiceStub(self._channel)
+        responses = client.set_some(request, metadata=self._metadata, timeout=timeout + len(request.values) * 0.1)
+        return [response.key async for response in responses]
+
+    @GRPCRequestHandler()
+    async def set_topology_decommissions(
+        self: CVClientProtocol,
+        device_ids: list[str],
+        workspace_id: str,
+        timeout: float = DEFAULT_API_TIMEOUT,
+    ) -> list[DecommissionConfig]:
+        """
+        Decommission devices using arista.studio_topology.v1.DecommissionConfigService.SetSome API.
+
+        Parameters:
+            device_ids: Device IDs to decommission.
+            workspace_id: Workspace ID to decommission the devices from.
+            timeout: Timeout in seconds.
+
+        Returns:
+            List of DecommissionConfig objects.
+        """
+        decommission_configs = [DecommissionConfig(key=DeviceKey(device_id=device_id, workspace_id=workspace_id)) for device_id in device_ids]
+        request = DecommissionConfigSetSomeRequest(decommission_configs)
+        client = DecommissionConfigServiceStub(self._channel)
+        responses = client.set_some(request, metadata=self._metadata, timeout=timeout)
+
+        return [response.key async for response in responses]
+
+    @GRPCRequestHandler()
+    async def get_all_decommissions(
+        self: CVClientProtocol,
+        device_ids: list[str],
+        workspace_id: str,
+        state: Literal["success", "unspecified", "failure"],
+        timeout: float = DEFAULT_API_TIMEOUT,
+    ) -> list[DecommissionConfig]:
+        """
+        Subscribe to the decommission status with arista.studio_topology.v1.DecommissionService.Subscribe API.
+
+        Parameters:
+            device_ids: Device IDs to decommission.
+            workspace_id: Workspace ID to decommission the devices from.
+            state: Decommission state to wait for.
+            timeout: Timeout in seconds.
+
+        Returns:
+            List of DecommissionConfig objects.
+        """
+        decommissioning_devices = [Decommission(key=DeviceKey(device_id=device_id, workspace_id=workspace_id)) for device_id in device_ids]
+        request = DecommissionStreamRequest(decommissioning_devices)
+        client = DecommissionServiceStub(self._channel)
+        responses = client.subscribe(request, metadata=self._metadata, timeout=timeout)
+        async for response in responses:
+            LOGGER.debug("wait_for_decommission_success: Response is '%s.'", response)
```

Comment:
```suggestion
            LOGGER.debug("wait_for_decommission_state: Response is '%s.'", response)
```
Or maybe `wait_for_decommission_complete`? This should also be the function name I think.